ts-client/src/amm/index.ts

```diff
@@ -87,66 +105,54 @@ const getRemainingAccounts = (poolState: PoolState) => {
   return accounts;
 };
 
-const getAccountsBuffer = async ({
-  connection,
-  vaultA,
-  vaultB,
-  apyPda,
-  poolState,
-}: {
-  connection: Connection;
-  vaultA: VaultState;
-  vaultB: VaultState;
-  apyPda: PublicKey;
-  poolState: PoolState;
-}) => {
-  return connection.getMultipleAccountsInfo([
-    apyPda,
-    vaultA.tokenVault,
-    vaultB.tokenVault,
-    vaultA.lpMint,
-    vaultB.lpMint,
-    poolState.aVaultLp,
-    poolState.bVaultLp,
-    poolState.lpMint,
-    SYSVAR_CLOCK_PUBKEY,
-  ]);
+type DecoderType = { [x: string]: (accountData: Buffer) => BN | ApyState };
+const decodeAccountTypeMapper = (type: AccountType): ((accountData: Buffer) => BN | ApyState) => {
+  const poolCoder = new BorshCoder(AmmIdl as Idl);
```

Comment:
i think we should bring this from outside so we dont need to reinitialize this on each call